### PR TITLE
Spring cleaning 🧼 

### LINKED ISSUE #730: Upkeep for pins
Pre-history

* [x] `usethis::use_readme_rmd()`
* [x] `usethis::use_roxygen_md()`
* [x] `usethis::use_github_links()`
* [x] `usethis::use_pkgdown_github_pages()`
* [x] `usethis::use_tidy_github_labels()`
* [ ] `usethis::use_tidy_style()`
* [x] `usethis::use_tidy_description()`
* [x] `urlchecker::url_check()`

2020

* [x] `usethis::use_package_doc()`
Consider letting usethis manage your `@importFrom` directives here.
`usethis::use_import_from()` is handy for this.
* [x] `usethis::use_testthat(3)` and upgrade to 3e, [testthat 3e vignette](https://testthat.r-lib.org/articles/third-edition.html)
* [x] Align the names of `R/` files and `test/` files for workflow happiness.
The docs for `usethis::use_r()` include a helpful script.
`usethis::rename_files()` may be be useful.

2021

* [ ] `usethis::use_tidy_dependencies()`
* [x] `usethis::use_tidy_github_actions()` and update artisanal actions to use `setup-r-dependencies`
* [x] Remove check environments section from `cran-comments.md`
* [x] Bump required R version in DESCRIPTION to 3.5
* [x] Use lifecycle instead of artisanal deprecation messages, as described in [Communicate lifecycle changes in your functions](https://lifecycle.r-lib.org/articles/communicate.html)
* [x] Make sure RStudio appears in `Authors@R` of DESCRIPTION like so, if appropriate:
`person("RStudio", role = c("cph", "fnd"))`

2022

* [x] `usethis::use_tidy_coc()`
* [x] Handle and close any still-open `master` --> `main` issues
* [x] Update README badges, instructions in [r-lib/usethis#1594](https://github.com/r-lib/usethis/issues/1594)
* [ ] Update errors to rlang 1.0.0. Helpful guides:
<https://rlang.r-lib.org/reference/topic-error-call.html>
<https://rlang.r-lib.org/reference/topic-error-chaining.html>
<https://rlang.r-lib.org/reference/topic-condition-formatting.html>
* [x] Update pkgdown site using instructions at <https://tidytemplate.tidyverse.org>
* [x] Ensure pkgdown `development` is `mode: auto` in pkgdown config
* [ ] Re-publish released site; see [How to update a released site](https://pkgdown.r-lib.org/dev/articles/how-to-update-released-site.html)
* [x] Update lifecycle badges with more accessible SVGs: `usethis::use_lifecycle()`

2023

Necessary:

* [x] Update copyright holder in DESCRIPTION: `person(given = "Posit Software, PBC", role = c("cph", "fnd"))`
* [x] Update email addresses *@rstudio.com -> *@posit.co
* [x] Update logo (https://github.com/rstudio/hex-stickers); run `use_tidy_logo()`
* [x] `usethis::use_tidy_coc()`
* [x] `usethis::use_tidy_github_actions()`

Optional:

* [x] Review 2022 checklist to see if you completed the pkgdown updates
* [x] Prefer `pak::pak("org/pkg")` over `devtools::install_github("org/pkg")` in README
* [ ] Consider running `use_tidy_dependencies()` and/or replace compat files with `use_standalone()`
* [x] `use_standalone("r-lib/rlang", "types-check")` instead of home grown argument checkers
* [x] Add alt-text to pictures, plots, etc; see https://posit.co/blog/knitr-fig-alt/ for examples

### PR DESCRIPTION
**Pull request by juliasilge**
Closes #730

### CODE REVIEW

**Comment by juliasilge** [comment]
Closes #682

File: R/pin-read-write.R [hunk 1]
```diff
@@ -252,39 +252,41 @@ hash_file <- function(path) {
...
   }
 }
-check_name <- function(x) {
+
+check_pin_name <- function(x, call = caller_env()) {
   if (grepl("\\\\|/", x, perl = TRUE)) {
-    abort("`name` must not contain slashes", class = "pins_check_name")
+    cli_abort(
+      "{.var name} must not contain slashes",
+      class = "pins_check_name",
```
Comment (DavisVaughan) [inline R/pin-read-write.R:268]: Very minor - We have somewhat gotten in the habit of classing our errors with

```
{pkg}_{error}_{descriptor}

# like
pins_error_invalid_name
```

in practice it doesn't matter much, just letting you know

File: R/pin-read-write.R [hunk 2]
```diff
@@ -252,39 +252,41 @@ hash_file <- function(path) {
...
     return()
   }
 
   pin_hash <- pin_hash(fs::path(meta$local$dir, meta$file))
   if (!is_prefix(hash, pin_hash)) {
-    abort(paste0(
-      "Specified hash '", hash, "' doesn't match pin hash '", pin_hash, "'"
-    ))
+    cli_abort(
+      "Specified hash {.val {hash}} doesn't match pin hash {.val {pin_hash}}",
```
Comment (DavisVaughan) [inline R/pin-read-write.R:288]:
```suggestion
      "Specified hash {.val {hash}} doesn't match pin hash {.val {pin_hash}}.",
```

Also very minor, we often try to end our error messages with periods as much as possible, just by convention

File: R/pin-read-write.R (hunk 1 above)
Comment (juliasilge) [reply R/pin-read-write.R:268]: Thanks! This particular one has been around for a while (it predates my work on pins) so I probably won't change it but I'll keep in mind moving forward. 👍

**Pull request merged**